src/evergreen/task.py

```diff
@@ -307,6 +309,24 @@ def is_timeout(self) -> bool:
             return self.status_details.timed_out
         return False
 
+    def is_setup_failure(self) -> bool:
+        """
+        Whether task is a setup failure.
+
+        :return: True if task is a setup failure.
+        """
+        if not self.is_success() and self.status_details and self.status_details.type:
+            return self.status_details.type == EVG_SETUP_FAILURE_STATUS_TYPE
+        return False
+
+    def is_completed(self) -> bool:
+        """
+        Whether task is completed.
+
+        :return: True if task is completed.
+        """
+        return self.status == EVG_SUCCESS_STATUS or self.status == EVG_FAILED_STATUS
```

Comment:
is system failure also considered to have status as "failure" or should we also check for EVG_SYSTEM_FAILURE_STATUS here?

Reply:
Got it !